numpy/core/src/umath/ufunc_object.c

```diff
@@ -1025,6 +1029,13 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
             }
 
             switch (str[0]) {
+                case 'a':
```

Comment:
Switch statements actually compile as hardwired binary search, so for small lists of possibilities the savings probably don't amount to much. I think the main virtues are compactness, clarity, an fall through, which can be used to select code entry points. The last is probably frowned upon by structured code fanatics...